lib/iris/util.py

```diff
@@ -2222,27 +2225,90 @@ def _strip_metadata_from_dims(cube, dims):
     return reduced_cube
 
 
-def mask_cube_from_shapefile(cube, shape, minimum_weight=0.0, in_place=False):
-    """Take a shape object and masks all points not touching it in a cube.
+def mask_cube_from_shapefile(
+    cube: iris.cube.Cube,
+    shape: shapely.Geometry,
+    shape_crs: cartopy.crs | pyproj.CRS,
+    in_place: bool = False,
+    minimum_weight: float = 0.0,
+    **kwargs,
+) -> iris.cube.Cube | None:
+    """Mask all points in a cube that do not intersect a shapefile object.
+
+    Notes
+    -----
+    .. deprecated:: 3.14
+
+        :func:`mask_cube_from_shapefile` function is scheduled for removal in a
+        future release, being replaced by :func:`iris.util.mask_cube_from_shape`,
+        which offers richer shape handling.
+    """
+    message = (
+        "iris.util.mask_cube_from_shapefile has been deprecated, and will be removed in a "
+        "future release. Please use iris.util.mask_cube_from_shape instead."
+    )
+    warn_deprecated(message)
+    # Call the new function `mask_cube_from_shape`with the same parameters.
+    return mask_cube_from_shape(
+        cube,
+        shape,
+        shape_crs,
+        in_place=in_place,
+        minimum_weight=minimum_weight,
+        **kwargs,
+    )
+
+
+def mask_cube_from_shape(
+    cube: iris.cube.Cube,
+    shape: shapely.Geometry,
+    shape_crs: cartopy.crs | pyproj.CRS,
+    in_place: bool = False,
+    minimum_weight: float = 0.0,
+    **kwargs,
+) -> iris.cube.Cube | None:
+    """Mask all points in a cube that do not intersect a shape object.
+
+    Mask a :class:`~iris.cube.Cube` with any shape object, (e.g. Natural Earth Shapefiles via cartopy).
+    Finds the overlap between the `shape` and the :class:`~iris.cube.Cube` and
+    masks out any cells that __do not__ intersect the shape.
 
-    Finds the overlap between the `shape` and the `cube` in 2D xy space and
-    masks out any cells with less % overlap with shape than set.
-    Default behaviour is to count any overlap between shape and cell as valid
+    Shapes can be polygons, lines or points.
+
+    By default, all cells touched by geometries are kept (equivalent to `minimum_weight=0`). This behaviour
+    can be changed by increasing the `minimum_weight` keyword argument or setting `all_touched=False`,
+    then only the only cells whose center is within the polygon or that are selected by Bresenham’s line algorithm
+    (for line type shapes) are kept.   For points, the `minimum_weight` is ignored, and the cell that intersects the point
+    is kept.
```

Comment:
```suggestion
    By default, all cells touched by geometries are kept (equivalent to ``minimum_weight=0``). This behaviour
    can be changed by increasing the ``minimum_weight`` keyword argument or setting ``all_touched=False``,
    then only the only cells whose center is within the polygon or that are selected by Bresenham’s line algorithm
    (for line type shapes) are kept.   For points, the `minimum_weight` is ignored, and the cell that intersects the point
    is kept.
```

If you want monospace: it's double-backtick.

If you want to refer to a parameter: I believe the standard recommends single-backtick.

Reply:
Fixed